master/buildbot/status/builder.py

```diff
@@ -322,12 +328,27 @@ def getLastFinishedBuild(self):
             b = self.getBuild(-2)
         return b
 
-    def setCategory(self, category):
+    def getTags(self):
+        return self.tags
+
+    def setTags(self, tags):
         # used during reconfig
-        self.category = category
+        self.tags = tags
+
+    def matchesAnyTag(self, tags):
+        return self.tags and any((tag in self.tags) for tag in tags)
+
+    def matchesAllTags(self, tags):
+        return self.tags and all((tag in self.tags) for tag in tags)
+
+    def setCategory(self, category):
+        return self.setTags([category])
 
     def getCategory(self):
-        return self.category
+        tags = self.getTags()
+        if tags:
+            return tags[0]
+        return None
```

Comment:
I'd add a deprecation warning here and in nine these can be just removed (if they are not already).